src/arrayMethodJoin.js

```diff
@@ -4,9 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let stringJoin = '';
+
+    if (this.length === 0) {
+      return stringJoin;
+    }
```

Comment:
not a single iteration will work, if length is 0
you will return stringJoin anyway, so this code is redundant